File: python/merge/_Merger_pdf_png.py
from Aspose.Imaging import Rectangle
from Aspose.Imaging.ImageOptions import PngOptions
from Aspose.Imaging.Sources import FileCreateSource
from Aspose.Imaging import Image
from aspose.pdf.Devices import PngDevice
from aspose.pdf.Devices import Resolution
from aspose.pdf import Document
import clr

aspose_pdf = clr.AddReference("../../lib/Aspose.PDF.dll")
aspose_imaging = clr.AddReference("../../lib/Aspose.Imaging.dll")

class pdf_to_png(object):
    def __init__(self, licence_path):
        self.dataDir = "../../TestData"
        if licence_path:
            self.licence_path = licence_path
            self.aspose_license = License()
            self.aspose_license.SetLicense(self.licence_path)

    def exec(self):
        pathSource = "../../TestData/test.pdf"

        # read pdf file to Aspose Document
        doc = Document(pathSource)

        # make list of path to temporary images
        images = []

        # pages in pdf counted from 1 to n
        for pageCount in range(1, doc.Pages.Count):
            # setup default resolution to pdf documents 72dpi
            resolution = Resolution(72)

            # create image device to save document as image with page dimensions and resolution
            imageDevice = PngDevice(
                doc.Pages[pageCount].PageInfo.Width, doc.Pages[pageCount].PageInfo.Height, resolution)

            outPath = "test_"+pageCount+".png"

            # process document page to image
            imageDevice.Process(doc.Pages[pageCount], outPath)
            images.Add(outPath)

        # make list pf parsed image sizes
        imageSizes = []
        for path in images:
            # load image from file, it supports a lot of formats
            image = Image.Load(path)
            imageSizes.Add(image.Size)

        newWidth = 0
        newHeight = 0
        for s in imageSizes:
            newWidth += s.Width
            if newHeight < s.Height:
                newHeight = s.Height
            else:
                newHeight = newHeight

        # use file system as source for save image
        # preserve image on the disk
        fileSource = FileCreateSource("./test.png", False)

        options = PngOptions
        options.Source = fileSource

        # create empty image with calculated width and hight
        newImage = Image.Create(options, newWidth, newHeight)
        stitchedWidth = 0
        for imagePath in images:
            image = Image.Load(imagePath)
            # create bounds to nsert small image into large
            bounds = Rectangle(stitchedWidth, 0, image.Width, image.Height)

            # combining images into new one
            # bounds = where to insert image
            # LoadArgb32Pixels = convert image chunk to 32bit Argb
            newImage.SaveArgb32Pixels(
                bounds, image.LoadArgb32Pixels(image.Bounds))
            stitchedWidth += image.Width

        # save created image to disk
        newImage.Save()

File: cpp/merge/_Merger_pdf_png.cpp
#include "Aspose.PDF.Cpp/Document.h"
#include "Aspose.PDF.Cpp/Generator/Drawing/Path.h"
#include "Aspose.PDF.Cpp/Devices/Resolution.h"
#include "Aspose.PDF.Cpp/Generator/Image.h"
#include "Aspose.PDF.Cpp/Devices/Device.h"
#include "Aspose.PDF.Cpp/Page.h"
#include "Aspose.PDF.Cpp/PageCollection.h"
#include "Aspose.PDF.Cpp/Devices/ImageDevice.h"
#include "Aspose.PDF.Cpp/Devices/PngDevice.h"
#include "Aspose.PDF.Cpp/Rectangle.h"

using namespace System;
using namespace Aspose::Pdf;

void pdf_to_png()
{
    auto pathSource = u"../../TestData/test.pdf";

    // read pdf file to Aspose Document
    auto doc = MakeObject<Document>(inputFilename);

    // make list of path to temporary images
    ArrayList<String> images = new ArrayList<>();

    // pages in pdf counted from 1 to n
    for (int pageCount = 1; pageCount <= doc.getPages().size(); pageCount++)
    {
        // setup default resolution to pdf documents 72dpi
        var resolution = new com.aspose.pdf.devices.Resolution(72);

        // create image device to save document as image with page dimensions and resolution
        var imageDevice = new com.aspose.pdf.devices.PngDevice(
            (int)doc.getPages().get_Item(pageCount).getPageInfo().getWidth(),
            (int)doc.getPages().get_Item(pageCount).getPageInfo().getHeight(),
            resolution);

        var outPath = "test_" + pageCount + ".png";

        // process document page to image
        imageDevice.process(doc.getPages().get_Item(pageCount), outPath);
        images.add(outPath);
    }

    // make list pf parsed image sizes
    ArrayList<com.aspose.imaging.Size> imageSizes = new ArrayList<>();
    for (var path : images)
    {
        // load image from file, it supports a lot of formats
        com.aspose.imaging.RasterImage image = (com.aspose.imaging.RasterImage)com.aspose.imaging.Image.load(path);
        imageSizes.add(image.getSize());
    }

    int newWidth = 0;
    int newHeight = 0;
    for (com.aspose.imaging.Size s : imageSizes)
    {
        newWidth += s.getWidth();
        newHeight = newHeight < s.getHeight() ? s.getHeight() : newHeight;
    }

    // use file system as source for save image
    com.aspose.imaging.Source fileSource = new com.aspose.imaging.sources.FileCreateSource(
        "./test.png",
        false); // preserve image on the disk

    com.aspose.imaging.imageoptions.PngOptions options = new com.aspose.imaging.imageoptions.PngOptions();
    options.setSource(fileSource);

    // create empty image with calculated width and hight
    var newImage = (com.aspose.imaging.fileformats.png.PngImage)com.aspose.imaging.Image.create(options, newWidth, newHeight);

    int stitchedWidth = 0;

    for (int i = 1; i < sizeof(images); i++)
    {
        auto fs = images[i];

        var image = (com.aspose.imaging.RasterImage)com.aspose.imaging.Image.load(imagePath);
        {
            // TODO: read image size

            // create bounds to nsert small image into large
            com.aspose.imaging.Rectangle bounds = new com.aspose.imaging.Rectangle(
                stitchedWidth,
                0,
                image.getWidth(),
                image.getHeight());

            // combining images into new one
            newImage.saveArgb32Pixels(
                bounds,                                     // where to insert image
                image.loadArgb32Pixels(image.getBounds())); // convert image chunk to 32bit Argb
            stitchedWidth += image.getWidth();
        }
    }

    // save created image to disk
    newImage.save();
}
